spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/RemoveCarryoverIterator.java

```diff
@@ -151,13 +153,21 @@ private int[] generateIndicesToIdentifySameRow(int columnSize) {
     return indices;
   }
 
-  private boolean isSameRecord(Row currentRow, Row nextRow) {
-    for (int idx : indicesToIdentifySameRow) {
+  protected boolean isSameRecord(Row currentRow, Row nextRow) {
+    for (int idx : indicesToIdentifySameRow()) {
       if (isDifferentValue(currentRow, nextRow, idx)) {
         return false;
       }
     }
 
     return true;
   }
+
+  protected StructType rowType() {
```

Comment:
Can we move these up to base class ?

Reply:
The class `ComputeUpdateIterator` doesn't need it. Can we do that until all subclasses of `ChangelogIterator` need it?

Reply:
Removed it.